tests/test_registry.py

```diff
@@ -645,6 +645,58 @@ def raise_exception():
         assert "error" in response
         assert random_val == response["error"]["data"]["random"]
 
+    def test_encoder_exception_batch(self):
+        def custom_encode(input):
+            if "foo" in str(input):
+                raise InternalError("Could not parse the input data.")
+            else:
+                return json.JSONEncoder().encode(input)
```

Comment:
I think this test would be more robust if you stored the original encoder and used it for the else case.